### DISCUSSION TITLE
time limit exceeded

### DISCUSSION THREAD

**Discussion by programmerarunkumar** [ImageMagick/ImageMagick#3965] [Help]
Getting java: time limit exceeded `Operation canceled' @ fatal/cache.c/GetImagePixelCache/1696. exception and server stops .
How we can fix this

**Comment by snibgo** [comment]
Your policy.xml file may contain a line like this:
```
  <policy domain="resource" name="time" value="10"/>
```
... which would limit processing time to 10 seconds. You can remove that line, or comment it out.

**Reply by programmerarunkumar** [reply]
Currently Having 300 as value for time
What this time actucally means (Maximum time for one image or all )?
What is Maximum time we can allocate
If the time exceeds , imagemagick wil go down ?

**Reply by programmerarunkumar** [reply]
> ... which would limit processing time to 10 seconds. You can remove that line, or comment it out.

Any way to do this from command line ?

**Comment by snibgo** [comment]
The limit is per command. If an IM command is running for more than that time, it will halt with an error.

I don't know what the maximum possible value is. "3600" (one hour) is fine. It seems to accept values up to 4294967295 and beyond without problem.

policy.xml is a text file, and can be edited with any text editor. I suppose it could be done with CLI tools such as sed, but I can't see the point.

**Reply by programmerarunkumar** [reply]
> The limit is per command. If an IM command is running for more than that time, it will halt with an error.

For Method How we can consider the limit
I'm using MagickWand Method to resize the Image 
     NewMagickWand
     MagickReadImage
     MagickResizeImage
     MagickWriteImages
     DestroyMagickWand

How we can consider limit here ? for each method or for all method of image resize

**Reply by programmerarunkumar** [reply]
> I don't know what the maximum possible value is. "3600" (one hour) is fine. It seems to accept values up to 4294967295 and beyond without problem.

Even after increasing the time getting java: time limit exceeded `Operation canceled' @ fatal/cache.c/GetImagePixelCache/1696.

**Comment by snibgo** [comment]
Looking through the source code, the time resource seems to be checked only in cache.c GetImagePixelCache(), but this is called (indirectly) by many higher-level processes. So I suspect that, of your list, the limit might be reached in MagickReadImage or MagickResizeImage or MagickWriteImages, but probably not in NewMagickWand or DestroyMagickWand.

The time that is actually measured seems to be from the first call to GetImagePixelCache() within this execution of the program.

**Comment by snibgo** [comment]
> Even after increasing the time getting java: time limit exceeded `Operation canceled' @ fatal/cache.c/GetImagePixelCache/1696.

What does "magick -list resource" say? How long did the command run before the failure?

**Reply by programmerarunkumar** [reply]
magick -list resource have Time: 1800

**Comment by snibgo** [comment]
So, does your program take more than 1800 seconds without completing?

**Reply by programmerarunkumar** [reply]
yes

**Comment by snibgo** [comment]
So, your program has not completed within 1800 seconds, and you have set a limit of 1800 seconds. As expected, the program is halted.

I suppose your program is doing a lot of work. Perhaps it has a bug, and is stuck in an infinite loop.

**Reply by programmerarunkumar** [reply]
Im just calling MagickWandGenesis method at the start 

and calling below 5 methods for resizing the image (maximum 5 threads will call at a time)

NewMagickWand
MagickReadImage
MagickResizeImage
MagickWriteImages
DestroyMagickWand


Getting java: time limit exceeded `Operation canceled' @ fatal/cache.c/GetImagePixelCache/1696. 
This exception is  due to MagickWandGenesis which will be called at the start ?  is it taking more time ?

**Comment by snibgo** [comment]
I suggest you create the smallest program that shows the problem. For example, if you think the probem occurs in MagickWandGenesis then nothing that occurs after that is relevant, so remove it from your program. 

When you have done that, show us the complete program. Also tell us how you compiled and built the program (include any CLI commands), and what version of IM you use, and what output the program made. If your program uses any input files, also show those.

One possibility is that your build process is wrong. For example, you might be compiling against one version of IM and linking to a different version.

**Reply by programmerarunkumar** [reply]
After not calling the MagickWandGenesis at the start it is working fine

Can we call the below method without calling MagickWandGenesis at start ?
NewMagickWand
MagickReadImage
MagickResizeImage
MagickWriteImages
DestroyMagickWand
